### PR TITLE
New hook HOOK_USESKILLON

### PR DESCRIPTION
**Pull request by FakelsHub**
There will be something incomprehensible, ask the question.

### CODE REVIEW

File: sfall/Modules/MiscPatches.cpp [hunk 1]
```diff
@@ -855,6 +873,21 @@ void KeepWeaponSelectModePatch() {
...
+	// Fixed getting distance from source to target when using skills
+	// Note: You need to use the Doctor/First-Aid skill on the player, this will cause the party member to apply his skill,
+	// but the engine function is not implemented completely so using this fix without overriding the Doctor/First-Aid
+	// functions in the global script is impractical, so this should be disabled by default
+	if (GetConfigInt("Misc", "PartyMemberSkillFix", 0) != 0) {
+		dlog("Applying party member using skill Doctor/First-Aid patch.", DL_INIT);
+		HookCall(0x412836, action_use_skill_on_hook);
+		dlogr(" Done", DL_INIT);
+	}
+	// Small code fix (change obj_dude to source)
```
Comment (NovaRain) [inline sfall/Modules/MiscPatches.cpp]: Does this fix address any vanilla bug or just for making the hook work properly?

Comment (FakelsHub) [reply sfall/Modules/MiscPatches.cpp]: For the hook, for the game itself there is no difference what to use source or obj_dude

**Comment by NovaRain** [comment]
Tested the hook a bit, but I'm not sure what PartyMemberSkillFix really does. I use the script to get Lenny to do the healing when I use First Aid/Doctor skill on myself:
```
procedure start begin
   if not init_hook then begin
      variable
         user := get_sfall_arg,
         target := get_sfall_arg,
         skill := get_sfall_arg;

      if (target == dude_obj) and (skill == SKILL_FIRST_AID or skill == SKILL_DOCTOR) then begin
         if Lenny_Ptr then begin
            display_msg("Lenny tries to heal you.");
            set_sfall_return(Lenny_Ptr);
         end else begin
            display_msg("Sorry, you should get Lenny to heal yourself.");
            set_sfall_return(-1);
         end
      end
   end
end
```
Lenny moved close to me and healed me no matter PartyMemberSkillFix is enabled or not. Could you give an example to show the difference?

**Comment by FakelsHub** [comment]
option and hook are not connected to each other.
need to use the option without a hook.

**Comment by NovaRain** [comment]
>option and hook are not connected to each other.
need to use the option without a hook.

From your description the option is only useful with a global script. What's the script if not using the UseSkillOn hook?

**Pull request closed without merging**